spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/ribbon/RibbonClientHttpRequestFactory.java

```diff
@@ -48,6 +48,9 @@ public RibbonClientHttpRequestFactory(SpringClientFactory clientFactory, LoadBal
 	public ClientHttpRequest createRequest(URI originalUri, HttpMethod httpMethod)
 			throws IOException {
         String serviceId = originalUri.getHost();
+        if (serviceId == null) {
+            throw new IOException("Invalid hostname in the URI [" + originalUri.toASCIIString() + "]");
```

Comment:
Using a serviceId w/ underscore was exploding way futher in the call stack on the ConcurrentMap, this at least makes it clear why it's failing. The problem here seems to be the mismatch between what you can register as a serviceId and the expectation that's a valid hostname here...

Can you point me to where I should change so it happens at startup ? I am a *real* noob at spring dev